### PR TITLE
RGB Text Boxes and Debugger Window Reorg

### PR DESCRIPTION
**Pull request by Zychronix**
Moved around a few things in the debug window and made it bigger. Added RGB options for the text box.  There is new transparency modes as well added for the change.

### CODE REVIEW

File: src/main/java/legend/core/gpu/Gpu.java [hunk 1]
```diff
@@ -1075,6 +1075,48 @@ public int handleTranslucence(final int x, final int y, final int texel, final T
...
       }
 
+      case FULL_BACKGROUND -> {
+        r = br;
+        g = bg;
+        b = bb;
+      }
+
+      case TQUATER_B_FOREGROUND -> {
+        r = Math.min(0xff, (int)Math.floor(br / 1.33) + fr);
```
Comment (LordMonoxide) [inline src/main/java/legend/core/gpu/Gpu.java:1085]: We can safely do this faster and more accurately with int math
`br * 3 / 4 + fr`

Comment (LordMonoxide) [reply src/main/java/legend/core/gpu/Gpu.java:1085]: Same as below too

File: src/main/java/legend/game/Scus94491BpeSegment_8002.java [hunk 2]
```diff
@@ -2704,17 +2704,80 @@ public static void renderTextboxBackground(final int textboxIndex) {
...
+              .rgb(0, Config.getTextBoxRgb(0))
+              .pos(0, x - textbox.width_1c, y - textbox.height_1e)
+              .rgb(1, Config.getTextBoxRgb(1))
+              .pos(1, x + textbox.width_1c, y - textbox.height_1e)
+              .rgb(2, Config.getTextBoxRgb(2))
+              .pos(2, x - textbox.width_1c, y + textbox.height_1e)
+              .rgb(3, Config.getTextBoxRgb(3))
+              .pos(3, x + textbox.width_1c, y + textbox.height_1e)
+            );
+          }else if(Config.getTextBoxColourMode() == 1) {
```
Comment (LordMonoxide) [inline src/main/java/legend/game/Scus94491BpeSegment_8002.java:2720]:
```suggestion
          } else if(Config.getTextBoxColourMode() == 1) {
```

File: src/main/java/legend/game/Scus94491BpeSegment_8002.java [hunk 3]
```diff
@@ -2704,17 +2704,80 @@ public static void renderTextboxBackground(final int textboxIndex) {
...
+              .rgb(0, Config.getTextBoxRgb(4))
+              .pos(0, x, y - textbox.height_1e)
+              .rgb(1, Config.getTextBoxRgb(5))
+              .pos(1, x + textbox.width_1c, y - textbox.height_1e)
+              .rgb(2, Config.getTextBoxRgb(6))
+              .pos(2, x, y + textbox.height_1e)
+              .rgb(3, Config.getTextBoxRgb(7))
+              .pos(3, x + textbox.width_1c, y + textbox.height_1e)
+            );
+          }else if(Config.getTextBoxColourMode() == 2) {
```
Comment (LordMonoxide) [inline src/main/java/legend/game/Scus94491BpeSegment_8002.java:2743]:
```suggestion
          } else if(Config.getTextBoxColourMode() == 2) {
```

File: src/main/java/legend/game/debugger/DebuggerController.java [hunk 4]
```diff
@@ -136,6 +195,46 @@ public void initialize() {
...
+    this.textBox8R.setValueFactory(new SpinnerValueFactory.IntegerSpinnerValueFactory(0, 255, (Config.getTextBoxRgb(7) & 0xff)));
+    this.textBox8G.setValueFactory(new SpinnerValueFactory.IntegerSpinnerValueFactory(0, 255, ((Config.getTextBoxRgb(7) >> 8) & 0xff)));
+    this.textBox8B.setValueFactory(new SpinnerValueFactory.IntegerSpinnerValueFactory(0, 255, ((Config.getTextBoxRgb(7) >> 16) & 0xff)));
+    this.textBoxTransparencyMode.setValueFactory(new SpinnerValueFactory.IntegerSpinnerValueFactory(0, 10, Config.getTextBoxTransparencyMode()));
+
+    if(Config.getTextBoxColourMode() == 0) {
+      textBoxBase.setSelected(true);
+      textBoxHorizontal.setSelected(false);
+      textBoxVertical.setSelected(false);
+    }else if(Config.getTextBoxColourMode() == 1) {
```
Comment (LordMonoxide) [inline src/main/java/legend/game/debugger/DebuggerController.java:229]:
```suggestion
    } else if(Config.getTextBoxColourMode() == 1) {
```

File: src/main/java/legend/game/debugger/DebuggerController.java [hunk 5]
```diff
@@ -136,6 +195,46 @@ public void initialize() {
...
+      textBoxBase.setSelected(false);
+      textBoxHorizontal.setSelected(true);
+      textBoxVertical.setSelected(false);
+    }else if(Config.getTextBoxColourMode() == 2) {
```
Comment (LordMonoxide) [inline src/main/java/legend/game/debugger/DebuggerController.java:233]:
```suggestion
    } else if(Config.getTextBoxColourMode() == 2) {
```

File: src/main/java/legend/game/debugger/DebuggerController.java [hunk 6]
```diff
@@ -389,4 +488,178 @@ private void toggleFastText(final ActionEvent event) {
...
+  }
+
+  @FXML
+  private void setTextBoxRadio(final ActionEvent event) {
+    if((event.toString().contains("textBoxBase"))) {
+      textBoxBase.setSelected(true);
+      textBoxHorizontal.setSelected(false);
+      textBoxVertical.setSelected(false);
+      Config.setTextBoxColourMode(0);
+    }else if (event.toString().contains("textBoxHorizontal")) {
```
Comment (LordMonoxide) [inline src/main/java/legend/game/debugger/DebuggerController.java:504]:
```suggestion
    } else if (event.toString().contains("textBoxHorizontal")) {
```

File: src/main/java/legend/game/debugger/DebuggerController.java [hunk 7]
```diff
@@ -389,4 +488,178 @@ private void toggleFastText(final ActionEvent event) {
...
+      textBoxBase.setSelected(false);
+      textBoxHorizontal.setSelected(true);
+      textBoxVertical.setSelected(false);
+      Config.setTextBoxColourMode(1);
+    }else if (event.toString().contains("textBoxVertical")) {
```
Comment (LordMonoxide) [inline src/main/java/legend/game/debugger/DebuggerController.java:509]:
```suggestion
    } else if (event.toString().contains("textBoxVertical")) {
```

**Review by LordMonoxide — CHANGES_REQUESTED** [review]

File: src/main/java/legend/core/gpu/Gpu.java (hunk 1 above)
Comment (Zychronix) [reply src/main/java/legend/core/gpu/Gpu.java]: I updated both TQUATER_B_FOREGROUND  and TQUARTER_B_TQUARTER_F to reflect the change.

File: src/main/java/legend/game/debugger/DebuggerController.java (hunk 6 above)
Comment (Zychronix) [reply src/main/java/legend/game/debugger/DebuggerController.java]: Fixed

File: src/main/java/legend/game/debugger/DebuggerController.java (hunk 7 above)
Comment (Zychronix) [reply src/main/java/legend/game/debugger/DebuggerController.java]: Fixed

File: src/main/java/legend/game/debugger/DebuggerController.java (hunk 5 above)
Comment (Zychronix) [reply src/main/java/legend/game/debugger/DebuggerController.java]: Fixed

**Pull request merged**